packages/front-end/src/app/components/EGPageHeader.vue

```diff
@@ -1,17 +1,21 @@
 <script setup lang="ts">
-  const props = withDefaults(
-    defineProps<{
-      title: string;
-      description?: string;
-      showBack?: boolean;
-      backButtonLabel?: string;
-      backAction: () => void;
-    }>(),
-    {
-      description: '',
-      showBack: true,
-    },
-  );
+  const props = defineProps<{
+    title: string;
+    description?: string;
+    showBack: boolean;
+    backButtonLabel?: string;
+    backAction?: () => void;
+  }>();
+
+  const router = useRouter();
+
+  const defaultBackAction = () => {
+    router.go(-1);
```

Comment:
navigating back causes some unintended behaviour if the page is navigated to from an unexpected source. for a quick example try:

- open new tab
- navigate to http://localhost:3000/labs/bbac4190-0446-4db4-a084-cfdbc8102297/4L3Qt66Y9SS4Y7?tab=Run+Details
- click back - it won't do anything or it will go to your new tab homepage, depending on browser

and then try
- navigate to http://localhost:3000/labs/bbac4190-0446-4db4-a084-cfdbc8102297/4L3Qt66Y9SS4Y7?tab=Run+Details
- switch between the tabs on the page a couple times
- click back
- it will switch between tabs because that action changes the url and adds a page to the router stack

I think we need to avoid popping pages from the router to go back, and push the intended back-to page instead - this is what I had to do to fix back-button-related problems last sprint

Reply:
Yep good call, I've made the action empty by default which can be overridden by an explicit path.